### DISCUSSION TITLE
Allow users to enter cheat days

### DISCUSSION THREAD

**Discussion by Dhavrus** [iSoron/uhabits#1238] [Feature Requests]
The frequency of some of my habits can change depending on rules I set. For instance, one of my habits is to exercise twice a week. However, on weeks I'm travelling, I allow myself to exercise once. The same principle applies to other habits depending on factors like if I'm working remotely, if I'm at the office, on vacation, etc. I'm sure other users have similar rules because there are habits that can only be done in specific circumstances (e.g at work, at home, etc.). 

When version 2.0 was released, I thought that I could use the skip days on habits that have rules. However, it seems that skip-days just give you some more time to do your habit. It doesn't replace actually checking a day. In the above example, if I'm travelling and I enter 1 checking day and 1 skip day, the streak will be broken and the score affected.

It would be nice to allow the user to enter cheat days. Doing so will allow users to maintain the streak and score when a habit is not performed due to rules or specific circumstances.

**Comment by hiqua** [comment]
* why not enter more skip days until you get the chance to properly check?
* why not just check your cheat day if you allow it to be the same anyway?

**Comment by ke352802081770314** [comment]
Personally I would dislike option 2 because I can no longer tell which weeks I've only done it once. Score the same does not mean it should look the same in the tracker.
I have daily habits and skip does not break the streak. So option 1 is probably what you are looking for. You will just have to enter multiple skip days in such a week.

Naively I would expect that scoring works like the skip day wasn't part of the calendar at all, so if I mark Wednesday as skip, after Tuesday Thursday follows immediately for the scoring logic. But that doesn't seem to be the case, at least for streaks, even the skip days are included in the count.

**Reply by oxc** [reply]
Yes, that's bugging me a little as well. I would like the skip days to be counted as fulfilled, or have the opportunity to mark days as cheat days (which would be like skip days but count as fulfilled for the streak).
Ideally, I would like to have a setting how many cheat days are allowed per month (e.g. 5 times a week + 2 cheat days per month). But perhaps that gets out of hand...
